Formula/arangodb.rb

```diff
@@ -54,29 +55,28 @@ def install
       system "go", "build", *std_go_args(ldflags: ldflags), "github.com/arangodb-helper/arangodb"
     end
 
-    mkdir "build" do
-      openssl = Formula["openssl@1.1"]
-      args = std_cmake_args + %W[
-        -DHOMEBREW=ON
-        -DCMAKE_BUILD_TYPE=RelWithDebInfo
-        -DUSE_MAINTAINER_MODE=Off
-        -DUSE_JEMALLOC=Off
-        -DCMAKE_SKIP_RPATH=On
-        -DOPENSSL_USE_STATIC_LIBS=On
-        -DCMAKE_LIBRARY_PATH=#{openssl.opt_lib}
-        -DOPENSSL_ROOT_DIR=#{openssl.opt_lib}
-        -DCMAKE_OSX_DEPLOYMENT_TARGET=#{MacOS.version}
-        -DTARGET_ARCHITECTURE=nehalem
-        -DUSE_CATCH_TESTS=Off
-        -DUSE_GOOGLE_TESTS=Off
-        -DCMAKE_INSTALL_LOCALSTATEDIR=#{var}
-      ]
+    openssl = Formula["openssl@1.1"]
+    args = std_cmake_args + %W[
+      -DHOMEBREW=ON
+      -DCMAKE_BUILD_TYPE=RelWithDebInfo
+      -DUSE_MAINTAINER_MODE=Off
+      -DUSE_JEMALLOC=Off
+      -DCMAKE_LIBRARY_PATH=#{openssl.opt_lib}
+      -DOPENSSL_ROOT_DIR=#{openssl.opt_lib}
+      -DCMAKE_OSX_DEPLOYMENT_TARGET=#{MacOS.version}
+      -DUSE_CATCH_TESTS=Off
+      -DUSE_GOOGLE_TESTS=Off
+      -DCMAKE_INSTALL_LOCALSTATEDIR=#{var}
+    ]
+    if build.bottle? && Hardware::CPU.intel?
+      args << "-DTARGET_ARCHITECTURE=#{MacOS.version.requires_nehalem_cpu? ? "nehalem" : "core"}"
```

Comment:
Not sure if this is accurate.
https://github.com/arangodb/arangodb/blob/v3.9.1/cmake/OptimizeForArchitecture.cmake#L311-L321
```
   if(TARGET_ARCHITECTURE STREQUAL "core")
      list(APPEND _march_flag_list "core2")
      list(APPEND _available_vector_units_list "sse" "sse2" "sse3")
   elseif(TARGET_ARCHITECTURE STREQUAL "merom")
      list(APPEND _march_flag_list "merom")
      list(APPEND _march_flag_list "core2")
      list(APPEND _available_vector_units_list "sse" "sse2" "sse3" "ssse3")
   elseif(TARGET_ARCHITECTURE STREQUAL "penryn")
      list(APPEND _march_flag_list "penryn")
      list(APPEND _march_flag_list "core2")
      list(APPEND _available_vector_units_list "sse" "sse2" "sse3" "ssse3")
```

Reply:
`-march=merom` doesn't exist AFAIK in LLVM or GCC, so I think it'll fall back to `core2`. Merom seems like the correct choice here as Yosemite supports Merom as the oldest and GCC's `-march=core2` for Linux enables SSSE3.

Reply:
I think `merom` causes issues on Linux/GCC
```
  checking whether the C compiler works... no
  configure: error: in `/tmp/arangodb-20220425-9957-w22x6t/ArangoDB-3.9.1/build/3rdParty/libunwind/v1.5':
  configure: error: C compiler cannot create executables
```

Previous `core` attempt failed at
```
/tmp/arangodb-20220424-9965-zgp7zb/ArangoDB-3.9.1/3rdParty/iresearch/core/utils/simd_utils.hpp:249:32: error: 'CombineShiftRightLanes' was not declared in this scope
    249 |       const auto delta = vec - CombineShiftRightLanes<3>(vec, prev);
        |                                ^~~~~~~~~~~~~~~~~~~~~~
```

I guess the difference between host arch and target arch are causing some issues. For now, may just revert to `nehalem` everywhere to make sure it isn't due to other changes, and perhaps leave it until a user reports it fails on their system.

Reply:
> Previous `core` attempt failed at

That's because it's trying to use SIMD for something it shouldn't. Not sure why that is but it's likely a bug.

> I think merom causes issues on Linux/GCC

This is suprising: `Performing Test check_cxx_compiler_flag__march_merom - Success`

`-march=merom` does not work locally for me so it should have fallen back to `-march=core2`.

Will need full logs to figure out why.